src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -54,14 +65,106 @@ public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext,
+        {
+            return GetVariantAsync<T, TContext>(feature, appContext, true, cancellationToken);
+        }
+
+        public ValueTask<T> GetVariantAsync<T>(string feature, CancellationToken cancellationToken)
+        {
+            return GetVariantAsync<T, object>(feature, null, false, cancellationToken);
+        }
+
+        private async ValueTask<T> GetVariantAsync<T, TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            if (feature == null)
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            FeatureVariant variant = null;
+
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider.GetFeatureDefinitionAsync(feature, cancellationToken).ConfigureAwait(false);
+
+            if (featureDefinition != null)
+            {
+                if (string.IsNullOrEmpty(featureDefinition.Assigner))
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.InvalidConfiguration,
+                        $"Invalid feature variant assigner name for the feature {feature}");
+                }
+
+                IFeatureVariantAssignerMetadata assigner = GetFeatureAssignerMetadata(featureDefinition.Assigner);
+
+                if (assigner == null)
+                {
+                    string errorMessage = $"The feature assigner '{featureDefinition.Assigner}' specified for feature '{feature}' was not found.";
+
+                    if (!_options.IgnoreMissingFeatureAssigners)
+                    {
+                        throw new FeatureManagementException(FeatureManagementError.MissingFeatureAssigner, errorMessage);
+                    }
+                    else
+                    {
+                        _logger.LogWarning(errorMessage);
+
+                        return default(T);
+                    }
+                }
+
+                var context = new FeatureVariantAssignmentContext()
+                {
+                    FeatureDefinition = featureDefinition
+                };
+
+                //
+                // IContextualFeatureVariantAssigner
+                if (useAppContext)
+                {
+                    ContextualFeatureVariantAssignerEvaluator contextualAssigner = GetContextualFeatureVariantAssigner(featureDefinition.Assigner, typeof(TContext));
+
+                    if (contextualAssigner != null)
+                    {
+                        variant = await contextualAssigner.AssignVariantAsync(context, appContext, cancellationToken).ConfigureAwait(false);
+                    }
+                }
+
+                //
+                // IFeatureVariantAssigner
+                if (assigner is IFeatureVariantAssigner featureVariantAssigner)
```

Comment:
Can we put this in the else clause for clearer logic?

```
if (useAppContext)
{
    ...
}
else
{
    // here
}
```

Reply:
Good suggestion, it actually ended up changing the behavior a bit but I think it's better now.